test-workflows/pipelines/generators/clusterloader/eks/tests/pipeline.yaml

```diff
@@ -0,0 +1,112 @@
+  - name: eks-version
+    default: "1.20"
+    description: The EKS version to install.
+  - name: region
+    description: The region where the cluster is in.
+    default: "us-west-2"
+  - name: zones
+    description: The zones where the cluster is in.
+    default: "us-west-2a,us-west-2b,us-west-2c"
+  - name: nodegroup-name
+    default: "linux-nodes"
+    description: The name of the nodegroup of the cluster.
+  - name: node-type
+    default: "m5.xlarge"
+    description: The type of the EC2 instaces for the nodegroup of the cluster.
+  - name: desired-nodes
+    default: "100"
+    description: The desired number of nodes in the cluster.
+  - name: min-nodes
+    default: "1"
+    description: The minimum number of nodes in the cluster.
+  - name: max-nodes
+    default: "100"
+    description: The maximum number of nodes in the cluster.
+  - name: giturl
+    description: "git url to clone the package"
+    default: https://github.com/kubernetes/perf-tests.git
+  - name: pods-per-node
+    description: "pod density"
+    default: "10"
+  - name: nodes-per-namespace
+    description:  "nodes per namespace to get created for load test "
+    default: "10"
+  - name: cl2-load-test-throughput
+    description: " throughput used for mutate operations"
+    default: "10"
+  tasks:
+  - name: create-eks-cluster
+    taskRef:
+      name: eks-cluster-create
+    params:
+      - name: cluster-name
+        value: '$(params.cluster-name)'
+      - name: eks-version
+        value: '$(params.eks-version)'
+      - name: region
+        value: '$(params.region)'
+      - name: zones
+        value: '$(params.zones)'
+      - name: nodegroup-name
+        value: '$(params.nodegroup-name)'
+      - name: node-type
+        value: '$(params.node-type)'
+      - name: desired-nodes
+        value: '$(params.desired-nodes)'
+      - name: min-nodes
+        value: '$(params.min-nodes)'
+      - name: max-nodes
+        value: '$(params.max-nodes)'
+    workspaces:
+      - name: secrets
+        workspace: secrets
+      - name: config    
+        workspace: config
+  - name: runloadtest
+    runAfter: [create-eks-cluster]
+    taskRef:
+      name: clusterloader2
+    params:
+      - name: giturl
+        value: '$(params.giturl)'
+      - name: pods-per-node
+        value: '$(params.pods-per-node)'
+      - name: nodes-per-namespace
+        value: '$(params.nodes-per-namespace)'
+      - name: cl2-load-test-throughput
+        value: '$(params.cl2-load-test-throughput)'
+    workspaces:
+      - name: source
+        workspace: source
+      - name: secrets
+        workspace: secrets
+      - name: config    
+        workspace: config
+      - name: results
+        workspace: results
+  - name: delete-eks-cluster
+    runAfter: [runloadtest]
+    taskRef:
+      name: eks-cluster-teardown
+    params:
+      - name: cluster-name
+        value: '$(params.cluster-name)'
+      - name: region
+        value: '$(params.region)'
+    workspaces:
```

Comment:
Seems like this should be the default for tekton (i.e. tasks use all of a pipeline's workspaces). Consider cutting an issue to make this the default.